### PR TITLE
Sample v3: Improve login experience

### PR DESCRIPTION
**Pull request by cardoso**
- Added configuration section with storage and region options
- Improve layout (separate low-level samples from high-level samples placeholders)
- Fix login being called when pressing cells other than samples
- Fix crash on login error

Also small change in SDK:
- Don't do anything when flush is enabled but local storage is not (flushing without local storage causes crash)

Screen Shot 2020-09-25 at 11 12 42

#no_changelog

### CODE REVIEW

File: Sources_v3/Database/DatabaseContainer.swift [hunk 1]
```diff
@@ -82,7 +82,7 @@ class DatabaseContainer: NSPersistentContainer {
         
         persistentStoreDescriptions = [description]
                 
-        if shouldFlushOnStart {
+        if shouldFlushOnStart, kind != .inMemory {
```
Comment (VojtaStavik) [inline Sources_v3/Database/DatabaseContainer.swift:85]: This needs to be tested. Can you take care of that @mantergo and push a commit with the missing test, please?

**Review by VojtaStavik — COMMENTED** [review]
Looks good 👍 I like the changes to the login screen. Just a couple of thoughts and questions:

  - We should decide, whether the sample app should be a test app for the developers to use and test, or a canonical example of how to use our SDK. If it's the latter, we probably don't need to expose the settings since they are not meant to be exposed?

 - Low Level Client in an internal name for the SDK and we probably shouldn't  mention it anywhere publicly. The official names are `StreamChat` and `StreamChat UI`.

 - I'm a little bit afraid that the examples themselves are getting lost in all the settings. Maybe we should change it? Having all the settings on a different screen (or at least at the bottom of this screen?) and have the examples listed as the main thing we want to sell?

File: Sources_v3/Database/DatabaseContainer.swift (hunk 1 above)
Comment (mantergo) [reply Sources_v3/Database/DatabaseContainer.swift:85]: Thanks for pinging, added test 🙂

**Comment by VojtaStavik** [comment]
@cardoso We just merged a PR that adds more locations, can you extend the list with Singapore and Sydney?

**Comment by cardoso** [comment]
@VojtaStavik all done.

About whether this is a test or canonical sample: I think it should be both, it's just too useful to have a configurable app to test all kinds of settings and reproduce certain use cases quickly. It already isn't 100% canonical because it has multiple sample inside.

But I can explore hiding some of the test stuff in other PRs. Probably move the credentials and configuration stuff to a new modal screen.

**Review by VojtaStavik — APPROVED** [review]
Perfect, thank you 👍 Can you please remove the (Low-Lever Client) mention from the sample names, too? Feel free to merge afterwards.

**Comment by cardoso** [comment]
@VojtaStavik oops. Missed those. Done 👍

**Pull request merged**